compiler/rustc_codegen_cranelift/src/base.rs

```diff
@@ -512,7 +514,7 @@ fn codegen_fn_body(fx: &mut FunctionCx<'_, '_, '_>, start_block: Block) {
                     );
                 }
 
-                let have_labels = if options.contains(InlineAsmOptions::NORETURN) {
+                let have_labels = if asm_macro.diverges(*options) {
```

Comment:
This should never be a naked function. We already handle naked functions specially above.

Reply:
you're right, but at the moment `naked_asm!` can occur outside of naked functions.

There isn't really a technical reason to disallow `naked_asm!` anywhere `asm!` is allowed, but there also isn't really much practical use because `asm!` could just be used instead. @Amanieu and others: thoughts?

Reply:
`naked_asm` should be specifically disallowed anywhere except in naked functions. It doesn't make sense in other contexts.